src/store/selectors/activities/index.ts

```diff
@@ -496,9 +556,31 @@ function addOperationToActivitiesList(
       return addFlagMemberOperation(existingData, operation);
     case OperationType.RESOLVE_FLAG_MEMBER:
       return addResolveFlagMemberOperation(existingData, operation);
+    case OperationType.GIVE:
+      if (operation.data.metadata) {
+        switch (operation.data.metadata.type) {
+          case GiveType.DIRECT_GIVE:
+            return addDirectGiveOperation(
+              existingData,
+              operation as DirectGiveOperation
+            );
+          case GiveType.TIP:
+            return addTipOperation(existingData, operation as TipGiveOperation);
+          default:
+            throw new Error(
+              `Unexpected GIVE operation data type. Operation: ${JSON.stringify(
+                operation,
+                null,
+                2
+              )}`
+            );
+        }
+      } else {
+        // TODO(tina): Migrate GIVE operations to have metadata block
+        return addIndependentOperation(existingData, operation);
```

Comment:
should this be `addDirectGiveOperation` instead?

Reply:
no, addDirectGiveOperation requires the metadata block

Reply:
oh